File: app/src/main/java/com/trien/dnanimation/MyViewPagerAdapter.java
package com.trien.dnanimation;

import android.content.Context;
import androidx.viewpager.widget.PagerAdapter;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by Troy on 26-Mar-18.
 */

class MyViewPagerAdapter extends PagerAdapter {

    private LayoutInflater layoutInflater;
    private Context context;
    private int[] layouts;

    public MyViewPagerAdapter(Context context, int[] layouts) {

        this.context = context;
        this.layouts = layouts;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {

        layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        View view = null;
        if (layoutInflater != null) {
            view = layoutInflater.inflate(layouts[position], container, false);
        }
        container.addView(view);
        Log.d("trienViewid", String.valueOf(view != null ? view.getId() : 0) + " " + layouts[position]);
        return view;
    }

    @Override
    public int getCount() {
        return layouts.length;
    }

    @Override
    public boolean isViewFromObject(View view, Object obj) {
        return view == obj;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {

        View view = (View) object;
        container.removeView(view);
    }
}

File: app/src/main/java/com/trien/dnanimation/MyViewPagerAdapter.kt
package com.trien.dnanimation

import android.content.Context
import androidx.viewpager.widget.PagerAdapter
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * Created by Troy on 26-Mar-18.
 */

internal class MyViewPagerAdapter(private val context: Context, private val layouts: IntArray) : PagerAdapter() {

    private var layoutInflater: LayoutInflater? = null

    override fun instantiateItem(container: ViewGroup, position: Int): Any {

        layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        var view: View? = null
        if (layoutInflater != null) {
            view = layoutInflater!!.inflate(layouts[position], container, false)
        }
        container.addView(view)
        Log.d("trienViewid", (view?.id ?: 0).toString() + " " + layouts[position])
        return view!!
    }

    override fun getCount(): Int {
        return layouts.size
    }

    override fun isViewFromObject(view: View, obj: Any): Boolean {
        return view === obj
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {

        val view = `object` as View
        container.removeView(view)
    }
}
